nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/bin/nifi.sh

```diff
@@ -284,18 +281,21 @@ run() {
     # all other commands will terminate quickly so want to just wait for them
 
     #setup directory parameters
-    BOOTSTRAP_LOG_PARAMS="-Dorg.apache.nifi.bootstrap.config.log.dir="\""${NIFI_LOG_DIR}"\"""
-    BOOTSTRAP_PID_PARAMS="-Dorg.apache.nifi.bootstrap.config.pid.dir="\""${NIFI_PID_DIR}"\"""
-    BOOTSTRAP_CONF_PARAMS="-Dorg.apache.nifi.bootstrap.config.file="\""${BOOTSTRAP_CONF}"\"""
+    BOOTSTRAP_LOG_PARAMS="-Dorg.apache.nifi.bootstrap.config.log.dir=\"${NIFI_LOG_DIR}\""
+    BOOTSTRAP_PID_PARAMS="-Dorg.apache.nifi.bootstrap.config.pid.dir=\"${NIFI_PID_DIR}\""
+    BOOTSTRAP_CONF_PARAMS="-Dorg.apache.nifi.bootstrap.config.file=\"${BOOTSTRAP_CONF}\""
 
     BOOTSTRAP_DIR_PARAMS="${BOOTSTRAP_LOG_PARAMS} ${BOOTSTRAP_PID_PARAMS} ${BOOTSTRAP_CONF_PARAMS}"
-
-    RUN_NIFI_CMD="cd "\""${NIFI_HOME}"\"" && exec ${sudo_cmd_prefix} "\""${JAVA}"\"" -cp "\""${BOOTSTRAP_CLASSPATH}"\"" -Xms12m -Xmx24m ${BOOTSTRAP_DIR_PARAMS}  org.apache.nifi.bootstrap.RunNiFi"
+    run_nifi_cmd="exec ${JAVA} -cp ${BOOTSTRAP_CLASSPATH} -Xms12m -Xmx24m ${BOOTSTRAP_DIR_PARAMS} org.apache.nifi.bootstrap.RunNiFi"
+    if [ -n "${run_as_user}" ]; then
+      # Provide SCRIPT_DIR to nifi-env for the run.as user
+      run_nifi_cmd="sudo -u ${run_as_user} sh -c 'SCRIPT_DIR=${SCRIPT_DIR} && . ${SCRIPT_DIR}/nifi-env.sh && ${run_nifi_cmd} $@'"
+    fi
 
     if [ "$1" = "start" ]; then
-        (eval $RUN_NIFI_CMD $@ &)
+        (eval "cd ${NIFI_HOME} && ${run_nifi_cmd}" &)
     else
-        eval $RUN_NIFI_CMD $@
+        eval "cd ${NIFI_HOME} && ${run_nifi_cmd}"
```

Comment:
yep, messed up the conflict resolution, thanks for catching.  will adjust and submit.